api/object.go

```diff
@@ -135,6 +145,46 @@ type (
 	}
 )
 
+func ParseObjectHeadResponseFrom(header http.Header) (HeadObjectResponse, error) {
```

Comment:
This seems a bit overkill to expose in this file considering that only the client needs it. `parseObjectResponseHeaders` might be a slightly better name since it's used for 2 endpoints and not just the `HEAD` one.